src/allmydata/test/web/test_root.py

```diff
@@ -36,65 +36,36 @@ class RenderSlashUri(unittest.TestCase):
     """
 
     def setUp(self):
-        self.request = DummyRequest(b"/uri")
-        self.request.fields = {}
-
-        def prepathURL():
-            return b"http://127.0.0.1.99999/" + b"/".join(self.request.prepath)
-
-        self.request.prePathURL = prepathURL
         self.client = Mock()
         self.res = URIHandler(self.client)
 
     def test_valid(self):
         """
         A valid capbility does not result in error
         """
-        self.request.args[b"uri"] = [(
+        query_args = {b"uri": [
             b"URI:CHK:nt2xxmrccp7sursd6yh2thhcky:"
             b"mukesarwdjxiyqsjinbfiiro6q7kgmmekocxfjcngh23oxwyxtzq:2:5:5874882"
-        )]
-        self.res.render_GET(self.request)
-
-    def test_invalid(self):
-        """
-        A (trivially) invalid capbility is an error
-        """
-        self.request.args[b"uri"] = [b"not a capability"]
-        response_body = self.res.render_GET(self.request)
-
-        soup = BeautifulSoup(response_body, 'html5lib')
-
-        assert_soup_has_tag_with_content(
-            self, soup, "title", "400 - Error",
+        ]}
+        response_body = self.successResultOf(
+            render(self.res, query_args),
         )
-        assert_soup_has_tag_with_content(
-            self, soup, "h1", "Error",
-        )
-        assert_soup_has_tag_with_content(
-            self, soup, "p", "Invalid capability",
+        self.assertNotEqual(
+            response_body,
+            "Invalid capability",
         )
 
-    @given(
```

Comment:
I dropped Hypothesis here because this is web test code, not capability parsing code, and Hypothesis was only being used to vary the capability string.  That would be a good test for the capability parsing code but I think it's wasteful here.

Also, I *thought* I was going to have to make this test return a Deferred and that's incompatible with Hypothesis.  It turned out that I didn't have to do that but I had already deleted the Hypothesis stuff at that point so I just left it deleted.